DNN Platform/Library/Data/PetaPoco/PetaPocoMapper.cs

```diff
@@ -114,17 +116,15 @@ public static void SetMapper<T>(IMapper mapper)
             _lock.EnterWriteLock();
             try
             {
-                if (Mappers.GetMapper(typeof (T), mapper) is StandardMapper)
+                if (Mappers.GetMapper(typeof (T), _defaultMapper) is StandardMapper)
```

Comment:
I'm not sure if I understand this logic 100% percent, could you explain how this statement would ever by false. 

If I am understanding the code correctly it tries to get the mapper on the entity of `T` and if there is no mapper on entity of `T` it uses the default mapper variable `_defaultMapper`. In our case it is just the `StandardMapper`.

Can you confirm?

---
Some general thoughts I had about this change which I don't think we can easily implement but would like to get us thinking about. At first look of this code my first instinct was how can we update it to use Dependency Injection instead of instantiating the IMapper at line 45 and re-using it. Since this method is a static method I understand why we are instantiating it, but I would love to start thinking about using Dependency Injection more and how we can update this code to use it. **This last bit is just some thoughts and we don't need to add Dependency Injection**
--

Reply:
It's a bit of bizarre code for sure. It took me a little bit to fully wrap my head around it. Initially in #2764 I was incorrect in my assumptions. I think now it's at least working as it was before that PR. 

Nothing should be using the StandardMapper, every `T` I threw at the function while attached to the debugger returned our custom `PetaPocoMapper`. It's a very bad way (IMO) of checking if our mapper is registered yet or not for the requested type. 

Basically, On the first call for each type, `T`, our custom mapper isn't registered yet. The call to GetMapper wouldn't find that type, and it would then return the _defaultMapper. In #2764 I was using our custome mapper type, so the check for `is StandardMapper` was always false and our custom mapper was never registered as the result. 

Chatting offline with @daguiler earlier about this, as he had basically the same questions. The idea I came up with was to introduce a new class called `NullMapper` or `NotYetRegisteredMapper` (which would replace the `is StandardMapper` check), and this would make things a bit easier to understand. It would also allow the StandardMapper to be used, which I'm not even sure is possible in our implementation, but as it is right now (and in all previous version) I think the code would error if T actually used the StandardMapper. 

As for Dependency Injection. For sure! I'd love to  work together on improving the platform with DI. Lets plan to get some of that into v9.4.1 or v10 (depending on any breaking changes, etc). At least v6 of PetaPoco is netstandard so that should help us quite a bit on the journey to .netcore support.